[nteract/web] Save/upload to Github and other improvements.

ramantehlan opened the pull request:
This PR is to save back changes to the repo or the fork. 

It will also make the following changes: 

- [x] Save back changes to the repo or the fork.


> Successor of PR #5167

ramantehlan commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -425,7 +607,7 @@ return (
                       },
                 cursorBlinkRate: 0,
-                mode: "python"
+                mode: "markdown"
```
I have added a TODO to identify the language when the file is opened and update this value.

willingc replied:
Let's add a comment above line 610 to mention the TODO.

ramantehlan commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -159,58 +357,46 @@ function addBuffer(e){
-     if (!forkFound)
-        createFork(org, repo)
-    })
+  const getFileType = (fileName: string) => {
```
This function is not complete, but it will be used to get file type and update the mode/language type in the editor.

ramantehlan commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -58,38 +56,244 @@ function getPath(params){
 }
 
 
+const listForks = (owner, repo) => {
```
These functions will be moved to a separate util folder since they all are used to work with Github API and are not part of the main component.

ramantehlan commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -159,58 +357,46 @@ function addBuffer(e){
+      setFileBuffer({})
+      console.log("Notification: Data Saved")
+      }).catch( (e) => {
+        console.log("Notification: Error in saving")
```
I am using the console as a placeholder, this can later be replaced with the notification component.

willingc replied:
Let's add a comment above lines with a console.log message similar to "TODO update when using notification system"

willingc commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -302,7 +487,7 @@ function addBuffer(e){
 
 
-    // We won't be following this logic, we will render the data from github and only send changes to binder  
+    // We won't be following this logic, we will render the data from github and only send changes to binder
```
Let's go ahead and remove the code if it is not being used. Thanks.

ramantehlan replied:
It will be used in the next step when will connect binder to nteract web/play. However, that comment is irrelevant now, so I will update it.

willingc commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -227,8 +413,7 @@ function addBuffer(e){
       fileList =  [[""]]
-      console.log("Repo Not found")
-      console.log(e)
+      console.log("Notification: Repo not found")
```
Add above line 416 with a message similar to "TODO update when using notification system"

captainsafia commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -58,38 +56,244 @@ function getPath(params){
+          repo,
+        }).then(() => {
+            resolve()
+      })
```
We should add a 'catch' error handler here.

captainsafia commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -58,38 +56,244 @@ function getPath(params){
+    repo: string,
+    branch: string = `master`,
+    buffer: {},
+    commitMessage: string = `Auto Commit`,
```
Nit: let's make this a little more helpful. For example: "Auto commit from nteract web"

captainsafia commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -58,38 +56,244 @@ function getPath(params){
+    buffer: {},
+    commitMessage: string = `Auto Commit`,
+) => {
+    return new Promise( async (resolve, reject) => {
```
I don't think you need to wrap this in a promise. You can just make the method async and await the calls inside a try-catch.

captainsafia commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -316,13 +501,13 @@ const dialogInputStyle = { width: "98%" }
         <Layout>
           <Head>
-             <title>nteract play: Run interactive code</title> 
+             <title>nteract play: Run interactive code</title>
```
This 'Head' component can probably be exported from another file and used here.

captainsafia commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -159,58 +357,46 @@ function addBuffer(e){
 
-    // TODO: Add a blob here with changes made and push them to github
+  // To save/upload data to github
+  const onSave = (event) => {
```
We can use async/await in this event handler.

captainsafia commented on applications/web/pages/p/[...params].tsx:
```diff
@@ -159,58 +357,46 @@ function addBuffer(e){
 
   function isNotebook (name: string) {
-    return name.includes(".ipynb") 
+    return name.includes(".ipynb")
```
Let's use endsWith for a stricter check.

ramantehlan commented:
@captainsafia and @willingc, Thank you for the reviews, I have addressed the comments and this PR is ready for rereview. : )

I also wanna point out that I have created just one TODO for all the replacements for console notification with actual notification.
